lib/Serge/Engine/Plugin/parse_rc.pm

```diff
@@ -86,7 +86,7 @@ sub parse {
         if ($orig_str) {
             my $str = $orig_str;
             $str =~ s/""/"/g;
-            $translated_str = &$callbackref($str, undef, $hint, undef, $lang);
+            $translated_str = &$callbackref($str, undef, $hint, undef, $lang, $idstr);
```

Comment:
Now, this is a proper fix, and I'll be happy to merge the PR with this single-line change.

As for the importer, right now it is just a copy-paste of t/engine.t, and requires writing some additional tests; it also doesn't seem to do the actual import as the reference translations table in your test is empty. Would be more interesting to somehow hook the importing mode right into `engine.t`, where it would first generate the translations (and check them against reference files, and then use importer to import translations back and see that the reference import database matches the generated one.

Reply:
Hi @iafan.

I updated the PR with another commit to fix more issues with import and parse_rc.
The translations table in the reference output are now populated correctly.

I agree that we should try to re-use some of the code in engine.t to consolidate the duplicate code this PR creates.
I'm not sure how to implement it as you suggested yet, but I'll try to at least get one step closer.

I'm wondering if you want me to create a new PR with just the changes in `Importer.pm` and `parse_rc.pm`?

Thank you.

Reply:
In regards to "use importer to import translations back and see that the reference import database matches the generated one.", do you expect there to be a new `reference-output` database to check against the import? For example, `.\t\data\engine\parse_rc\00\reference-output\imported-database`? The `properties` and `translations` tables seem to be the only ones that are different.